internal/policies/apparmor/apparmor.go

```diff
@@ -0,0 +1,268 @@
+// Package apparmor provides a manager to apply apparmor policies.
+package apparmor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/ubuntu/adsys/internal/decorate"
+	log "github.com/ubuntu/adsys/internal/grpc/logstreamer"
+	"github.com/ubuntu/adsys/internal/i18n"
+	"github.com/ubuntu/adsys/internal/policies/entry"
+	"golang.org/x/exp/slices"
+)
+
+// Manager prevents running multiple apparmor update processes in parallel while parsing policy in ApplyPolicy.
+type Manager struct {
+	apparmorDir       string
+	apparmorParserCmd string
+
+	muMu    sync.Mutex             // protect pathsMu
+	pathsMu map[string]*sync.Mutex // mutex is per destination path (e.g. machine, users/user1, users/user2)
+}
+
+type options struct {
+	apparmorParserCmd string
+}
+
+// Option reprents an optional function to change the apparmor manager.
+type Option func(*options)
+
+// New creates a manager with a specific apparmor directory.
+func New(apparmorDir string, opts ...Option) (m *Manager, err error) {
```

Comment:
Right now, there's no way for this function to return an error. Even though this keeps the same signature as the other managers.New funcs, I think it's best to take a look at it and, if the function can't fail in any way, it shouldn't expect to return an error.

Reply:
I have no strong opinion either way on this. Both approach works for me (being future proof and return an error, or no error being returned and be able to update both API and package manager tests as needed). As this is an internal API, I would leave that up to your decision :)

I wouldn’t check `apparmor_parser` executable there yet, as you say, we aren’t sure we have a policy to apply, and the executable may vanish between the manager creation and the `ApplyPolicy` call.